worlds/zillion/__init__.py

```diff
@@ -393,7 +407,8 @@ def fill_slot_data(self) -> ZillionSlotInfo:  # json of WebHostLib.models.Slot
         # TODO: tell client which canisters are keywords
         # so it can open and get those when restoring doors
 
-        self.slot_data_ready.wait()
+        # `self.zz_system.get_game()` requires that item locations have been finalized.
+        self.finalize_item_locations_thread_safe()
         assert self.zz_system.randomizer, "didn't get randomizer from generate_early"
         game = self.zz_system.get_game()
```

Comment:
```suggestion
        game = self.finalize_item_locations_thread_safe().zz_game
```
I was repeating some work.